AirCasting/SessionViews/DefaultDeleteSessionViewModel.swift

```diff
@@ -0,0 +1,64 @@
+// Created by Lunar on 16/07/2021.
+//
+
+import Foundation
+
+class DefaultDeleteSessionViewModel: DeleteSessionViewModel {
+    
+    func isNotMicrophoneToggle() {
+        isMicrophone = false
+    }
+    
+    func isMicrophoneToggle() {
+        isMicrophone = true
+    }
+    
+    var isMicrophone: Bool = false {
+        willSet {
+            objectWillChange.send()
+        }
+    }
+    
+    var deleteEnabled: Bool = false {
+        willSet {
+            objectWillChange.send()
+        }
+    }
+    
+    func deleteSelected() {
+        #warning("TODO: Implement me")
+    }
+    
+    func didSelect(option: DeleteSessionOptionViewModel) {
+        guard let index = options.firstIndex(where: { $0.id == option.id }) else {
+            assertionFailure("Unknown option index")
+            return
+        }
+        options[index].toggleSelection()
+        if index == 0 {
+            for i in options.indices {
+                options[i].changeSelection(newSelected: options[0].isSelected)
+            }
+        }
+    }
+    
+    // It is variable because of the following scenario :
+    // 1. Go to delete session scene
+    // 2. Remove one of the streams on different device
+    // 3. It should after sync it should get removed from the view
+    
+    var options: [DeleteSessionOptionViewModel] {
+        willSet {
+            objectWillChange.send()
+        }
+    }
+    
+    init() {
+        #warning("TODO: When implementing please provide real data")
+        options =  !isMicrophone ? [.init(id: 0, title: "All", isSelected: false, isEnabled: false),
```

Comment:
now I see that `isMicrophone` is only used here to mock. I thought it's really affecting a UI, not only list of streams. I don't think we need it in that case. Since this task is only a view development we can put static mock data here for now. The way it will work when connecting to real data is that we'll be getting some object here representing session and we'll take streams from there and format them accordingly. We don't really need to know what are they (there is a question on how to display single-stream sessions, but that's out of scope here I think). Or am I wrong?

Reply:
You are right I think, will delete this